compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2140,7 +2140,7 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
 
     given MethodTypeMethods: MethodTypeMethods with
       extension (self: MethodType)
-        def isErased: Boolean = self.isErasedMethod
+        def isErased: Boolean = self.hasErasedParams
```

Comment:
This change is not right, it should probably return false as in the new scheme method types are not erased themself. `isErased` should be deprecated in `Quotes.scala`.

We have some changes in semantics of:
* `MethodTypeMethods.isErased`: Should be deprecated and replaced with `erasedParams` and maybe `hasErasedParams`
* `TermParamClauseMethods.isErased`: Should be deprecated and replaced with `erasedArgs` and maybe `hasErasedArgs`
* `TypeReprMethods.isErasedFunctionType`: we could make it work by detecting the refined function type. Though it might be cleaner to deprecate and add a more consistent API.
* `defn.FunctionClass`: needs to be deprecated. Should not have had default parameters in the first palce.

We can only deprecate these methods in a minor release.

What we should include in this PR are the new replacement as `@experimental` methods.

Reply:
And started the work on `defn.FunctionClass` in #16849

Reply:
```suggestion
        def isErased: Boolean = false
```
same for `TermParamClauseMethods.isErased`.

Reply:
We could change the documentation of `TypeReprMethods.isErasedFunctionType` in `Quotes.scala` to be 
```scala
        /** Is this type a function type with erased parameters?
        *
        *  @see `isFunctionType`
        */
```

Reply:
> * `MethodTypeMethods.isErased`: Should be deprecated and replaced with `erasedParams` and maybe `hasErasedParams`
> 
> * `TermParamClauseMethods.isErased`: Should be deprecated and replaced with `erasedArgs` and maybe `hasErasedArgs`
> 
> * `TypeReprMethods.isErasedFunctionType`: we could make it work by detecting the refined function type.

This is what I did, alongside with crashing `FunctionClass(isErased = true)` and adding `ErasedFunctionClass` (that just returns the `scala.compiletime.ErasedFunction` symbol).